src/nvs.rs

```diff
@@ -157,15 +157,14 @@ impl NvsEncrypted {
             None
         };
 
-        let c_key_partition = c_key_partition
```

Comment:
Good catch!

Mentioning below jfyi, your solution would work as well:
Don'we just need one extra `.as_ref()` here, right after `c_key_partition`?
Unless I don't understand Rust, the validity of the original, un-shadowed `c_key_partition` should be all the way down to the end of the function? It is just that when `map`-ing it, I accidentally unwrap an `Option<CString>` instead of an `Option<&CString>` thus destroying the value.

The reason wjhy Rust did not err on that that is because `p.as_ptr()` returns a raw pointer and not really a reference, so Rust does not give a damn that this pointer immediately becomes invalid.

Reply:
Yes,  I think if we `ref` the value, we could've left the code as it was.
Both seem to work as you describe.

Thanks!
```
 let c_key_partition = match c_key_partition { =>
     Some(ref v) => v.as_ptr(), => &CString
     None => core::ptr::null(),
 };
 unsafe {
       esp_partition_find_first(.., .., c_key_partition); 
 }
```

```
let c_key_partition = c_key_partition => *const i8, Option<CString>
    .as_ref() => Option<&CString>
    .map(|p| p.as_ptr()) => Option<*const i8>, &CString
    .unwrap_or(core::ptr::null()); <- (default)
unsafe {
       esp_partition_find_first(.., .., c_key_partition); 
}
```